_source/_posts/2022-01-27-build-website-with-eleventy.md

```diff
@@ -0,0 +1,334 @@
+ - A readme to demonstrate the usage of markdown (`README.md`)
+ - A page that is under authentication (`protected.html`)
+
+First, open the `index.html` file you just created and replace the code with the following.
+
+```html
+---
+layout: _layout_unauthenticated.11ty.js
+---
+
+<div class="row align-items-md-stretch">
+    <div class="col-md-12">
+        <div class="h-100 p-5 text-white bg-secondary rounded-3">
+            <h2>Using Okta with 11ty</h2>
+            <p>A small tutorial on writing static sites with 
+                <a class="text-dark" href="https://www.11ty.dev/" target="_blank">11ty</a>, 
+                secured by <a class="text-dark" href="https://www.okta.com/" target="_blank">Okta</a>
+                written by <a class="text-dark" href="https://profile.fishbowlllc.com/" target="_blank">Nik Fisher.</a>
+            </p>
+            <a class="btn btn-outline-light" role="button" href="Login">Login Now</a>
+            <a class="btn btn-light" role="button" href="README">README</a>
+        </div>
+    </div>
+</div>
+```
+
+You will notice the `layout` variable at the top of the file. You have not added this file to your project yet and will need to do that later. But you can assign the layout page to this file, so you don't have to add it later.
+
+Next, create a file for `login.html` and add the following code.
+
+```html
+---
+layout: _layout_unauthenticated.11ty.js
+---
+
+<div id="okta-login-container"></div>
+<script>
+  oktaSignIn.authClient.token.getUserInfo().then(function(user) {
+    window.location='/protected';
+  },function(error) {
+    oktaSignIn.showSignInToGetTokens({
+      el: '#okta-login-container'
+    }).then(function(tokens) {
+      oktaSignIn.authClient.tokenManager.setTokens(tokens);
+      oktaSignIn.remove();
+      window.location='/protected';
+    }).catch(function(err) {
+      console.error(err);
+    });
+  });
+</script>
+```
+
+The only HTML in this page here is a `div` with `id="okta-login-container"`. You call the `OktaSignIn` features and display the login screen to the user using the JavaScript code on this page. Okta will create the login form and handle the actual authentication. All you need to do is call the `showSignInToGetTokens` method and handle the promise to set the token using the `tokenManager`. Once you have done that, you will redirect the user to the `protected` page. 
+
+You can add the `protected.html` page now. Add the following code to it.
+
+```html
+---
+layout: _layout_authenticated.11ty.js
+---
+
+<div id="content"></div>
+
+<script>
+    function loadPage(){
+        //replace with API call 
+        document.getElementById('content').innerHTML = "<h2>Congrats</h2><h5>You've reached a protected page</h5>"
+    }
+</script>
+```
+
+Since you haven't seen the `_layout_authenticated.11ty.js` page yet, this will need some explanation. After verifying the user has is authenticated, the `loadPage` function is called by the layout.  
```

Comment:
```suggestion
Since you haven't seen the `_layout_authenticated.11ty.js` page yet, this will need some explanation. After verifying that the user has authenticated, the `loadPage` function is called by the layout.  
```
typo